### PR TITLE
[doc] Remove OpenOCD installation from "Getting Started"

### PR DESCRIPTION
**Pull request by dmcardle**
This PR primarily reworks the OpenOCD instructions, but it also includes assorted bug fixes.

### CODE REVIEW

File: doc/guides/getting_started/src/using_openocd.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+We use the regular upstream version of OpenOCD rather than the [RISC-V downstream fork](https://github.com/riscv/riscv-openocd).
+
+The Bazel-built OpenOCD binary is exposed as a Bazel target: `//third_party/openocd:openocd_bin`.
+It's not directly runnable, so we provide a runnable wrapper: `//third_party/openocd`.
+
+OpenOCD ships with a library of config files.
+Instead of using whichever config files happen to be installed on the system, users should prefer Bazel targets that expose config files directly from OpenOCD's source tree.
+Currently, we only expose OpenTitan's default JTAG adapter config as `//third_party/openocd:jtag_adapter_cfg`.
+
+```console
```
Comment (HU90m) [inline doc/guides/getting_started/src/using_openocd.md:16]:
```suggestion
```sh
```

Nit: I prefer the use `sh` (or `bash` if bash-isms used).

 I tend to reserve `console` for when wanting to show output:

````markdown
```console
$ python -c 'print("hello")'
hello
```
````

**Review by HU90m — APPROVED** [review]
This looks great, thanks for updating the docs!

File: doc/guides/getting_started/src/using_openocd.md (hunk 1 above)
Comment (dmcardle) [reply doc/guides/getting_started/src/using_openocd.md:16]: Ah, I have the same preference, but I opted to match the file. It uses `console` in a bunch of places that probably should be `sh`. WDYT?

If I get a chance today, I may append a commit that replaces all relevant `console`s with `sh`.

Comment (dmcardle) [reply doc/guides/getting_started/src/using_openocd.md:16]: Sorry, that comment didn't make sense — there's only one code block in this file. I was thinking of `setup_fpga.md`. Making this change now.

**Review by a-will — APPROVED** [review]

**Pull request merged**